source/hpack.c

```diff
@@ -783,14 +786,16 @@ int aws_hpack_resize_dynamic_table(struct aws_hpack_context *context, size_t new
     return AWS_OP_ERR;
 }
 
-void aws_hpack_set_max_table_size(struct aws_hpack_context *context, size_t new_max_size) {
-
-    if (!context->dynamic_table_size_update.pending) {
-        context->dynamic_table_size_update.pending = true;
+void aws_hpack_set_max_table_size(struct aws_hpack_context *context, size_t setting_max_size, bool update_table_size) {
+    if (update_table_size) {
```

Comment:
I cannot figure out the purpose of the `update_table_size` variable?

the `dynamic_table_size_update` struct is only used by the "encoder" part of `aws_hpack_context` (hence, my comment about it being confusing that encoder/decoder share a class)